boards/BMS/Src/Io/Io_CellTemperatures.c

```diff
@@ -0,0 +1,306 @@
+ * @param rx_raw_thermistor_voltage The buffer containing the raw thermistor
+ * voltages read from the thermistors
+ * @return EXIT_CODE_OK if the PEC15 check was successful. Else,
+ * EXIT_CODE_ERROR
+ */
+static ExitCode Io_CellTemperatures_ParseThermistorVoltagesAndPerformPec15Check(
+    size_t   current_chip,
+    size_t   current_register_group,
+    uint8_t *rx_raw_thermistor_voltages);
+
+/**
+ * Read raw thermistor voltages from the cell monitoring chips
+ * @return EXIT_CODE_OK if raw thermistor voltages are read successfully from
+ * all cell monitoring chips. Else, EXIT_CODE_ERROR
+ */
+static ExitCode Io_CellTemperatures_ReadRawThermistorVoltages(void);
+
+static ExitCode Io_CellTemperatures_ParseThermistorVoltagesAndPerformPec15Check(
+    size_t   current_chip,
+    size_t   current_register_group,
+    uint8_t *rx_raw_thermistor_voltages)
+{
+    size_t raw_thermistor_voltages_index = current_chip * NUM_OF_RX_BYTES;
+
+    for (size_t current_cell = 0U;
+         current_cell < NUM_OF_THERMISTORS_PER_REGISTER_GROUP; current_cell++)
+    {
+        const uint16_t raw_thermistor_voltage = (uint16_t)(
+            (rx_raw_thermistor_voltages[raw_thermistor_voltages_index]) |
+            ((rx_raw_thermistor_voltages[raw_thermistor_voltages_index + 1]
+              << 8)));
+
+        size_t curr_column =
+            current_cell +
+            current_register_group * NUM_OF_THERMISTORS_PER_REGISTER_GROUP;
+        if (current_register_group == AUX_REGISTER_GROUP_C)
+        {
+            // Subtract curr_column to ignore the reference voltage read back
+            // from aux register group B.
+            curr_column--;
+        }
+        raw_thermistor_voltages[current_chip][curr_column] =
+            (uint16_t)raw_thermistor_voltage;
+
+        // Each aux measurement is represented by 2 bytes. Therefore,
+        // the raw thermistor voltage index is incremented by 2 to retrieve the
+        // next thermistor voltage.
+        raw_thermistor_voltages_index += 2U;
+    }
+
+    const uint16_t received_pec15 = (uint16_t)(
+        (rx_raw_thermistor_voltages[raw_thermistor_voltages_index] << 8) |
+        (rx_raw_thermistor_voltages[raw_thermistor_voltages_index + 1]));
+
+    // Calculate the PEC15 using the first 6 bytes of data received from the
+    // chip.
+    const uint16_t calculated_pec15 = Io_LTC6813_CalculatePec15(
+        &rx_raw_thermistor_voltages[current_chip * NUM_OF_RX_BYTES], 6U);
+
+    return (received_pec15 == calculated_pec15) ? EXIT_CODE_OK
+                                                : EXIT_CODE_ERROR;
+}
+
+static ExitCode Io_CellTemperatures_ReadRawThermistorVoltages(void)
+{
+    uint16_t aux_register_group_cmd;
+    uint8_t  tx_cmd[NUM_OF_CMD_BYTES];
+    uint8_t  rx_thermistor_resistances
+        [NUM_OF_RX_BYTES * NUM_OF_CELL_MONITOR_CHIPS] = { 0 };
+
+    // The command used to start auxiliary (GPIO) measurements.
+    const uint16_t ADAX = 0x460 + (MD << 7) + CHST;
```

Comment:
![image](https://user-images.githubusercontent.com/27713668/99870666-d5a7f400-2b89-11eb-8efe-58d983f02bc5.png)
This is the command being called right?

Reply:
I'm just wondering why it has CHG rather than CHST

Reply:
Great catch. Luckily both values were set to 0